client/state/plugins/installed/test/selectors-ts.ts

```diff
@@ -213,3 +229,108 @@ describe( 'getAllPluginsIndexedBySiteId', () => {
 		expect( pluginsOne ).toBe( pluginsTwo );
 	} );
 } );
+
+describe( 'getPluginsOnSites', () => {
+	it( 'returns an object with plugins that are on the specified sites', () => {
+		const plugins = [ jetpackWithSites, akismetWithSites ];
+
+		const result = getPluginsOnSites( state, plugins );
+
+		expect( result ).toEqual( {
+			jetpack: jetpackWithSites,
+			akismet: akismetWithSites,
+		} );
+	} );
+
+	it( 'returns an empty object if the plugins have no sites', () => {
+		const plugins = [
+			{ ...jetpackWithSites, ...{ sites: {} } },
+			{ ...akismetWithSites, ...{ sites: {} } },
+		];
+
+		const result = getPluginsOnSites( state, plugins );
+
+		expect( result ).toEqual( {} );
+	} );
```

Comment:
Are we missing a test with an empty array of plugins, and/or with incorrect plugin slugs?

Reply:
I added tests for these cases.